typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,604 @@
+
+            const destAbsFilepath: string = path.join(this.enclaveDataVolMountpoint, destFilepathRelativeToEnclaveVolRoot);
+            
+            const promiseStatDestAbsFilepath: Promise<ResultAsync<fs.Stats, Error>> = new Promise((resolve, _unusedReject) => {
+                fs.stat(destAbsFilepath, (error: Error, response: fs.Stats) => {
+                    if (error === null) {
+                        resolve(okAsync(response));
+                    } else {
+                        resolve(errAsync(error));
+                    }
+                })
+            });
+            const resultStatDestAbsFilepath: Result<fs.Stats, Error> = await promiseStatDestAbsFilepath;
+            if (!resultStatDestAbsFilepath.isOk()) {
+                return err(new Error("The Kurtosis API asked us to copy static file " + staticFileId + " to path " + destFilepathRelativeToEnclaveVolRoot + 
+                " in the enclave volume which means that an empty file should exist there, " + "but no file exists at that path - this is a bug in Kurtosis!"));
+            }
+
+            
+            var srcFp: number;
+            var destFp: number;
+            try {
+                
+                const promiseOpenSrcFp: Promise<ResultAsync<number, Error>> = new Promise((resolve, _unusedReject) => {
+                    fs.open(srcAbsFilepath, 'r', (error: Error, response: number) => {
+                        if (error === null) {
+                            resolve(okAsync(response));
+                        } else {
+                            resolve(errAsync(error));
+                        }
+                    })
+                });
+                const resultOpenSrcFp: Result<number, Error> = await promiseOpenSrcFp;
+                if (!resultOpenSrcFp.isOk()) {
+                    return err(resultOpenSrcFp.error);
+                }
+                srcFp = resultOpenSrcFp.value;
+
+                try {
+                    const promiseOpenDestFp: Promise<ResultAsync<number, Error>> = new Promise((resolve, _unusedReject) => {
+                        fs.open(destAbsFilepath, 'w', (error: Error, response: number) => {
+                            if (error === null) {
+                                resolve(okAsync(response));
+                            } else {
+                                resolve(errAsync(error));
+                            }
+                        })
+                    });
+                    const resultOpenDestFp: Result<number, Error> = await promiseOpenDestFp;
+                    if (!resultOpenDestFp.isOk()) {
+                        return err(resultOpenDestFp.error);
+                    }
+                    destFp = resultOpenDestFp.value;
+
+                    try {
+                        const promiseCopyFile: Promise<void> = fsPromises.copyFile(srcAbsFilepath, destAbsFilepath);
+
+                        //TODO - REMOVE the following commented code
```

Comment:
This commented-out code is actually right. As-is, the current code will create a promise which completes.... at some later point. This means that if an error was thrown, we'll never detect it